sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosContainer.java

```diff
@@ -339,6 +339,25 @@ private TransactionalBatchResponse blockBatchResponse(Mono<TransactionalBatchRes
         return getCosmosPagedIterable(this.asyncContainer.queryItems(querySpec, options, classType));
     }
 
+    /**
+     * Query items in the current container returning the results as {@link CosmosPagedIterable}.
+     *
+     * @param <T> the type parameter.
+     * @param querySpec the query spec.
+     * @param options the options.
+     * @param classType the class type.
+     * @param feedRange the feedrange
+     * @return the {@link CosmosPagedIterable}.
+     */
+    @Beta(value = Beta.SinceVersion.V4_12_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public <T> CosmosPagedIterable<T> queryItems(SqlQuerySpec querySpec,
+                                                 CosmosQueryRequestOptions options,
+                                                 Class<T> classType,
+                                                 FeedRange feedRange) {
```

Comment:
ditto on moving FeedRange to `CosmosQueryRequestOptions.setFeedRange()`

Reply:
we can leave this discussion for later after this PR as this is Beta API, we can change it anytime later.

IMO, `FeedRange` should be part of the `CosmosQueryRequestOptions`. a method with 4 args is very confusing and overkill.